Projects/Features/SearchFeature/Sources/ViewControllers/AfterSearchViewController.swift

```diff
@@ -46,21 +44,56 @@ public final class AfterSearchViewController: TabmanViewController, ViewControll
     public static func viewController(
         afterSearchContentComponent: AfterSearchContentComponent,
         containSongsFactory: ContainSongsFactory,
-        viewModel: AfterSearchViewModel
+        reactor: AfterSearchReactor
     ) -> AfterSearchViewController {
         let viewController = AfterSearchViewController.viewController(storyBoardName: "Search", bundle: Bundle.module)
-        viewController.viewModel = viewModel
         viewController.afterSearchContentComponent = afterSearchContentComponent
         viewController.containSongsFactory = containSongsFactory
+        viewController.reactor = reactor
         return viewController
     }
 
     deinit {
         DEBUG_LOG("❌ \(Self.self)")
     }
+
+    public func bind(reactor: AfterSearchReactor) {
+        bindState(reacotr: reactor)
+        bindAction(reactor: reactor)
+    }
 }
 
 extension AfterSearchViewController {
+    func bindState(reacotr: AfterSearchReactor) {
+        let currentState = reacotr.state.share(replay: 2)
+
+        // TODO: 검색 결과 화면 나올 때 , Content쪽 tableView hidden처리 및 indicator start 시점 고려
+        currentState.map(\.dataSource)
+            .withUnretained(self)
```

Comment:
```suggestion
        currentState.map(\.dataSource)
            .filter { !$0.isEmpty }
            .withUnretained(self)
```
bind 안에서 filter하기보다 오퍼레이터로 필터하는게 좋을거같아요